### PR TITLE
misc fixes

### PR DESCRIPTION
**Pull request by ceyusa**
I tried to generate the sys bindings using the latest version of gir and I hit with some compilation errors. These patches address those errors (I think).

### CODE REVIEW

**Comment by sdroege** [comment]
These crates are unused but erroneously added by gir :) please remove them from lib.rs or even better fix gir to not add them at all

**Comment by ceyusa** [comment]
As far as I understand, all the external libraries in Gir_Gst*.toml are turn into "external crates ..." in lib.rs
if they are not used why are they declared in Gir_Gst*.toml? some linking stage that I'm not seeing?

Thanks for your patience :)

**Comment by sdroege** [comment]
Can they just be removed there? I think I added them there because the .gir file had them

**Comment by sdroege** [comment]
You have to tell me when you add new commits, I didn't get a notification for that because GitHub :(

With this, does regeneration of the -sys bindings cause any changes? Can you add another commit with the regeneration?

**Comment by sdroege** [comment]
For the gir-files update, can you sync them with the ones in gstreamer-rs?

File: gstreamer-app-sys/Cargo.toml [hunk 1]
```diff
@@ -1,45 +1,46 @@
...
-v1_2 = []
-v1_4 = ["v1_2"]
-v1_6 = ["v1_4"]
-v1_8 = ["v1_6"]
-v1_10 = ["v1_8"]
+dox = []
+v1_10 = ["v1_6"]
 v1_12 = ["v1_10"]
 v1_14 = ["v1_12"]
-dox = []
```
Comment (sdroege) [inline gstreamer-app-sys/Cargo.toml]: This needs some manual work. I'd suggest keeping the Cargo.toml as they were before with a simple `git checkout`.

Problem is that gir only adds version features for versions that have changes (for consistency I prefer to have them for all major versions), and doesn't add version features for minor releases (even if they add new API).

File: gstreamer-app-sys/build.rs [hunk 2]
```diff
@@ -22,14 +22,8 @@ fn find() -> Result<(), Error> {
         "1.12"
     } else if cfg!(feature = "v1_10") {
         "1.10"
-    } else if cfg!(feature = "v1_8") {
-        "1.8"
     } else if cfg!(feature = "v1_6") {
         "1.6"
-    } else if cfg!(feature = "v1_4") {
```
Comment (sdroege) [inline gstreamer-app-sys/build.rs]: And same here in the build.rs

**Review by sdroege — COMMENTED** [review]
Otherwise looks good

**Comment by ceyusa** [comment]
> For the gir-files update, can you sync them with the ones in gstreamer-rs?

https://github.com/sdroege/gstreamer-rs/pull/129

Is that what you asked or I misunderstand you? :S

**Pull request merged**